src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -213,6 +238,64 @@ public long getCurrentSpoolerSize() {
     public SpoolerConfig getSpoolConfig() {
         return config;
     }
-}
 
+    /**
+     * Extract message ids from diskQueueOfIds (received from on-disk database) and insert the message.
+     * ids into queueOfMessageId
+     */
+    private void persistentQueueSync(List<Long> diskQueueOfIds) {
+        if (!diskQueueOfIds.isEmpty()) {
+            long highestId = -1;
+            int i;
+            for (i = 0; i < diskQueueOfIds.size(); i += 1) {
+                //Check for queue space and remove if necessary
+                long currentId = diskQueueOfIds.get(i);
+                SpoolMessage message = spooler.getMessageById(currentId);
```

Comment:
Possible NPE here, `spooler` may not have been instantiated (setup).

Reply:
Thank you for catching this, I will fix it for the next commit